src/main/java/dev/andrylat/vomelianchuk/CardNumberOperations/CardNumberProcess.java

```diff
@@ -0,0 +1,118 @@
+package dev.andrylat.vomelianchuk.CardNumberOperations;
+
+import dev.andrylat.vomelianchuk.CardNumberOperations.enums.Errors;
+import dev.andrylat.vomelianchuk.CardNumberOperations.enums.PaymentSystem;
+
+import java.util.Optional;
+
+public class CardNumberProcess {
+    private static final int LENGTH_CARD_NUMBER = 16;
+    private static final String HIATUS_SEPARATOR = "\\s";
+    private static final int TEN = 10;
+
+    private String cardNumber;
```

Comment:
During implementation, the best functionality is when you call method, provide everything it needs and receive result. I mean that it is good to avoid unnecessary states. 
You should not use them there where we can avoid them.

Here you can create one public method and put there everything and receive result. 

Let's think, how it can be achieved.
What is needed as input? only card number. It means that you can create method "operate" and as parameter you can put card number.
As a result it can be payment system. Because you definitely need it. 
```
public PaymentSystem determinePaymentSystem(String cardNumber)
```

And you can have reasonable questions: what about errors? 
We discussed that you can throw exception if validation is failed. And don't forget that exception is a class and it means that you can store there aby data you want. 
So, put there list of errors. 
you should have something like 
```
class YourException extends RuntimeException {
  private List<String> errorMessages = ...;
  public YourException(String message, List<String> errorMessages) {
  ...
  }
  ...
}
```
And you have very easy use cases:
- if everything is ok, you will receive payment system.
- if no, you will catch exception with all errors.

And the most important part! 
If you make as I said, your code will be safe thread and it can work in multythreading env very good without any needs to create new object each time.
It is my main goal to you to explain it.

Reply:
Fixed here: 1783555